### PR TITLE
more fixes on integration

### PR DESCRIPTION
**Pull request by AntaresS**
Signed-off-by: Anda Xu <anda.xu@docker.com>

**- What I did**
- skipped new logging tests when running remotely
- moved `TestInspectInvalidReference ` to docker/cli
- fixed typo

**- How I did it**

**- How to verify it**
- run the whole integration testing suite on e2e

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by AntaresS** [comment]
cc @thaJeztah @tiborvass Hope we can get this in ASAP to unblock integrating with e2e.

File: integration-cli/docker_cli_inspect_test.go
Comment (tiborvass) [inline integration-cli/docker_cli_inspect_test.go]: What is the issue here?

Comment (AntaresS) [reply integration-cli/docker_cli_inspect_test.go]: @tiborvass from `17.06.2-ee-11` engine, it's not returning the complete error message. 
```
[]
Error: No such object: FooBar
```
vs
```
[]
Error response from daemon: no such image: FooBar: invalid reference format: repository name must be lowercase
```
Plus this test case seems to be a duplicate from `TestInspectUnknownObject` above. After discussing with @thaJeztah, we decide to disable it for now.

Comment (tiborvass) [reply integration-cli/docker_cli_inspect_test.go]: How could those two tests pass on any version of docker ? Aren't they mutually exclusive?

Comment (thaJeztah) [reply integration-cli/docker_cli_inspect_test.go]: From the other pull request https://github.com/moby/moby/pull/37003#discussion_r187942411 :

This test looks flawed; it doesn't test for the right part of the error message, which is at the end:

```
Error response from daemon: no such image: FooBar: invalid reference format: repository name must be lowercase
```

But there's actually a bug: `docker inspect` stops after this, and because of that masks other, valid, objects:

```
docker volume create FooBar
FooBar

docker inspect FooBar
[]
Error response from daemon: no such image: FooBar: invalid reference format: repository name must be lowercase

docker run -dit --name FooBar busybox
docker inspect FooBar 

# (outputs container inspect)
```

What's needed is;

- Test should check for `invalid reference format: repository name must be lowercase`
- Test should do `docker image inspect` (i.e. this error is specific to inspecting images)
- (not for this PR); we should fix `docker inspect` stopping at the first error it encounters to prevent it from masking valid objects

File: integration-cli/docker_cli_inspect_test.go
Comment (thaJeztah) [inline integration-cli/docker_cli_inspect_test.go]: Does it work if you use `docker image inspect` ?

```go
out, _, err := dockerCmdWithError("image", "inspect", "FooBar")
```

**Comment by tiborvass** [comment]
@AntaresS @thaJeztah We need to fix the bug introduced by https://github.com/moby/moby/commit/ebcb7d6b406fe50ea9a237c73004d75884184c33#diff-a715d1887671bdf0072480ba7d9773d2R466

The original version of the test was correct (checked with Tonis). That test if corrected would be rightfully failing on master: `docker inspect FooBar` should output `No such object`. Unfortunately, the fix is in docker cli. I'm suggesting we take this opportunity to migrate the test to docker/cli.

**Comment by AntaresS** [comment]
@tiborvass Done - https://github.com/docker/cli/pull/1071

File: integration-cli/docker_cli_inspect_test.go
Comment (cpuguy83) [inline integration-cli/docker_cli_inspect_test.go]:
```
$ docker inspect FooBar
[]
Error response from daemon: no such image: FooBar: invalid reference format: repository name must be lowercase
```

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

File: integration-cli/docker_cli_inspect_test.go
Comment (AntaresS) [reply integration-cli/docker_cli_inspect_test.go]: @cpuguy83 Discussed with @tiborvass @tonistiigi, this error should not be returned with detailed `no such image: FooBar: invalid reference format` message. Hence, it's already fixed in docker/cli#1071

Comment (AntaresS) [reply integration-cli/docker_cli_inspect_test.go]: Actually, this test case will fail because the current release does not include the fix from cli yet. Can we still disable this test for now? @tiborvass @thaJeztah

Comment (tiborvass) [reply integration-cli/docker_cli_inspect_test.go]: https://github.com/docker/cli/pull/1072 is merged, so I suggest we simply remove this test.

Comment (AntaresS) [reply integration-cli/docker_cli_inspect_test.go]: @tiborvass resolved

**Comment by tiborvass** [comment]
@AntaresS we should create a class of errors for "invalid" errors that can be considered "not found" and use that in the CLI. Can be done in a followup.

LGTM

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
Experimental failed on a flaky test;

```
18:43:42 --- FAIL: TestCreateServiceSecretFileMode (32.30s)
18:43:42 	daemon.go:285: [d85dc59262e7f] waiting for daemon to start
18:43:42 	daemon.go:317: [d85dc59262e7f] daemon started
18:43:42 	create_test.go:190: timeout hit after 30s: task count at 1 waiting for 0
18:43:42 	daemon.go:275: [d85dc59262e7f] exiting daemon
```

**Comment by thaJeztah** [comment]
We can probably merge this without waiting for CI, as it only removes a test, and skips another one @tiborvass

**Pull request merged**
